AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+DeleteUser.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+#if COCOAPODS
+import AWSMobileClient
+#else
+import AWSMobileClientXCF
+#endif
+
+extension AuthenticationProviderAdapter {
+
+    func deleteUser(request: AuthDeleteUserRequest, completionHandler: @escaping (Result<Void, AuthError>) -> Void) {
+        // By default, ASWMobileClient calls signOut internally during deleteUser.
+        // For Amplify, we instead call Amplify's signOut function, which contains some higher level logic.
+        awsMobileClient.deleteUser(signOut: true) { [weak self] error in
+            guard let error = error else {
+                if let self = self, self.userdefaults.isPrivateSessionPreferred() {
+                    // Reset the user defaults.
+                    self.userdefaults.storePreferredBrowserSession(privateSessionPrefered: false)
+                }
+
+                completionHandler(.success(()))
+                Amplify.Hub.dispatch(to: .auth, payload: HubPayload(eventName: HubPayload.EventName.Auth.signedOut))
```

Comment:
Should we dispatch deletedUser event here instead? I am not sure what we have decided across team to be send here when a delete succeed

Reply:
Yes, we discussed this between all three platforms and emitting both userDeleted and signedOut hub events was the agreed behavior.  The rational for firing both was that in many cases, for top-level state within the app, they may not care whether the user deleted their account any only want to update state on signedIn vs signedOut.